File: 1334/c/c.cc
#include <climits>
#include <cstdio>
#include <iostream>
#include <vector>

using namespace std;
typedef long long ll;

int main() {
  int t;
  scanf("%d", &t);
  string out;
  while (t--) {
    int n;
    scanf("%d", &n);
    vector<ll> a(n), b(n);
    for (int i = 0; i < n; ++i)
      scanf("%lld%lld", &a[i], &b[i]);
    vector<ll> rest(n);
    ll sum = 0;
    for (int i = 0; i < n; ++i)
      sum += rest[i] = max(0ll, a[i] - b[(i - 1 + n) % n]);
    ll ans = LLONG_MAX;
    for (int i = 0; i < n; ++i)
      ans = min(ans, sum - rest[i] + a[i]);
    out += to_string(ans) + "\n";
  }
  printf("%s", out.c_str());
}

File: 1334/c/c.py
t = int(input())
for case_num in range(t):
    n = int(input())
    a = []
    b = []
    for i in range(n):
        ai, bi = map(int, input().split(' '))
        a.append(ai)
        b.append(bi)
    rest = []
    for i in range(n):
        rest.append(max(a[i] - b[(i - 1 + n) % n], 0))
    s = sum(rest)
    ans = int(1e18)
    for i in range(n):
        ans = min(ans, s - rest[i] + a[i])
    print(ans)
